packages/sanity/src/core/hooks/useTimeZone.tsx

```diff
@@ -0,0 +1,334 @@
+import {DATE_FORMAT} from '../scheduledPublishing/constants'
+import {type NormalizedTimeZone} from '../scheduledPublishing/types'
+import {debugWithName} from '../scheduledPublishing/utils/debug'
+import getErrorMessage from '../scheduledPublishing/utils/getErrorMessage'
+
+enum TimeZoneEvents {
+  update = 'timeZoneEventUpdate',
+}
+
+export type TimeZoneScopeType = 'scheduledPublishing' | 'contentReleases' | 'input'
+
+type NoIdRequiredTypes = 'scheduledPublishing' | 'contentReleases'
+type IdRequiredTypes = 'input'
+
+type ReleasesOrScheduledPublishingScope = {
+  type: NoIdRequiredTypes
+  defaultTimeZone?: string
+}
+
+type InputScope = {
+  type: IdRequiredTypes
+  id: string
+  defaultTimeZone?: string
+  relativeDate?: Date // offsets can change over time, so we use a relative date to get the offset
+}
+
+export type TimeZoneScope = ReleasesOrScheduledPublishingScope | InputScope
+
+const debug = debugWithName('useScheduleOperation')
+
+export const timeZoneLocalStorageNamespace = 'timeZone.'
+
+const timeZoneCache = new Map<string, NormalizedTimeZone[]>()
+const offsetCache = new Map<
+  string,
+  {abbreviation: string; alternativeName: string; offset: string}
+>()
+
+const offsetToMinutes = (offset: string): number => {
+  if (!offset) return 0
+  const multiplier = offset.startsWith('-') ? -1 : 1
+  if (!offset.includes(':')) return multiplier * Number(offset.replace(/[+-]/, '')) * 60
+
+  const [hours, minutes] = offset.replace(/[+-]/, '').split(':').map(Number)
+  return multiplier * (hours * 60 + minutes)
+}
+
+function getCachedTimeZoneInfo(
+  locale: string,
+  canonicalIdentifier: string,
+  relativeDateForZones?: Date,
+): {abbreviation: string; alternativeName: string; offset: string} {
+  const dateKey = relativeDateForZones?.toISOString().slice(0, 10) || 'now'
+  const cacheKey = `${locale}_${canonicalIdentifier}_${dateKey}`
+  // relative date is not used for caching, but it's used to invalidate the cache when it changes
+  if (offsetCache.has(cacheKey)) {
+    return offsetCache.get(cacheKey)!
+  }
+
+  const formatter = new Intl.DateTimeFormat(sanitizeLocale(locale), {
+    timeZone: canonicalIdentifier,
+    timeZoneName: 'long',
+  })
+
+  const shortFormatter = new Intl.DateTimeFormat(sanitizeLocale(locale), {
+    timeZone: canonicalIdentifier,
+    timeZoneName: 'short',
+  })
+
+  const dateToUse = relativeDateForZones ?? new Date()
+  const parts = formatter.formatToParts(dateToUse)
+  const shortParts = shortFormatter.formatToParts(dateToUse)
+  const rawOffset = formatInTimeZone(dateToUse, canonicalIdentifier, 'xxx')
+  // If the offset is +02:00 then we can just show +2, if it has +13:45 then we should show +13:45, remove the leading +0 and just leave a + if a number under 10, remove the :00 at the end
+  const offset = rawOffset
+    .replace(/([+-])0(\d)/, '$1$2')
+    .replace(/([+-])0$/, '$1')
+    .replace(/:00$/, '')
+    .replace(/[+]0$/, '')
```

Comment:
Is this related to formatting? Or something different? 🤔 
Trying ot understand why this needs to be here 🤔

Reply:
This is related to formatting, we do this here. Apparently some timezones are off by minutes and not just hours 🙄